april_tag_updates/new_april.py

```diff
@@ -0,0 +1,151 @@
+import imutils
+import cv2
+ARUCO_DICT = {
+	"DICT_4X4_50": cv2.aruco.DICT_4X4_50,
+	"DICT_4X4_100": cv2.aruco.DICT_4X4_100,
+	"DICT_4X4_250": cv2.aruco.DICT_4X4_250,
+	"DICT_4X4_1000": cv2.aruco.DICT_4X4_1000,
+	"DICT_5X5_50": cv2.aruco.DICT_5X5_50,
+	"DICT_5X5_100": cv2.aruco.DICT_5X5_100,
+	"DICT_5X5_250": cv2.aruco.DICT_5X5_250,
+	"DICT_5X5_1000": cv2.aruco.DICT_5X5_1000,
+	"DICT_6X6_50": cv2.aruco.DICT_6X6_50,
+	"DICT_6X6_100": cv2.aruco.DICT_6X6_100,
+	"DICT_6X6_250": cv2.aruco.DICT_6X6_250,
+	"DICT_6X6_1000": cv2.aruco.DICT_6X6_1000,
+	"DICT_7X7_50": cv2.aruco.DICT_7X7_50,
+	"DICT_7X7_100": cv2.aruco.DICT_7X7_100,
+	"DICT_7X7_250": cv2.aruco.DICT_7X7_250,
+	"DICT_7X7_1000": cv2.aruco.DICT_7X7_1000,
+	"DICT_ARUCO_ORIGINAL": cv2.aruco.DICT_ARUCO_ORIGINAL,
+	"DICT_APRILTAG_16h5": cv2.aruco.DICT_APRILTAG_16h5,
+	"DICT_APRILTAG_25h9": cv2.aruco.DICT_APRILTAG_25h9,
+	"DICT_APRILTAG_36h10": cv2.aruco.DICT_APRILTAG_36h10,
+	"DICT_APRILTAG_36h11": cv2.aruco.DICT_APRILTAG_36h11
+}
+left_right_dist = 25.5
+cap = cv2.VideoCapture(0,cv2.CAP_DSHOW)
+cap2 = cv2.VideoCapture(1,cv2.CAP_DSHOW)
+arucoDict = cv2.aruco.Dictionary_get(ARUCO_DICT["DICT_APRILTAG_36h11"])
+arucoParams = cv2.aruco.DetectorParameters_create()
+CACHED_PTS = None
+CACHED_IDS = None
+Line_Pts = None
+measure = None
+CACHED_PTS2 = None
+CACHED_IDS2 = None
+Line_Pts2 = None
+measure2 = None
+window = 'Camera'
+cv2.namedWindow(window)
+window2='Camera2'
+cv2.namedWindow(window2)
+while True:
+    Dist = []
+    ret, image = cap.read()
+    image = imutils.resize(image, width=800)
+    (corners, ids, rejected) = cv2.aruco.detectMarkers(
+        image, arucoDict, parameters=arucoParams)
+    if len(corners) <= 0:
+        if CACHED_PTS is not None:
+            corners = CACHED_PTS
+    if len(corners) > 0:
+        CACHED_PTS = corners
+        if ids is not None:
+           ids = ids.flatten()
+           CACHED_IDS = ids
+        else:
+           if CACHED_IDS is not None:

```

Comment:
for each branch, can we add the logic of what we are checking and why it matters?